pkg/controller/petset/pet.go

```diff
@@ -216,6 +220,7 @@ func (p *apiServerPetClient) Update(pet *pcb, expectedPet *pcb) (updateErr error
 		if p, getErr = pc.Get(pod.Name); getErr != nil {
 			return getErr
 		}
+		time.Sleep(retrySleepTime)
```

Comment:
BTW, it seems that there is a bug here,
https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/petset/pet.go#L212-L216

when first update fails, we get a `new` pod(which is different the pod previously want to be updated). So, I think we need to `copyPetID` before update retries.

@bprashanth 

What do you think about it? I will have a deeper look and try to mock a test case.